nomos-cli/src/cmds/disseminate/mod.rs

```diff
@@ -0,0 +1,217 @@
+use clap::{Args, ValueEnum};
+use crossterm::execute;
+use full_replication::{AbsoluteNumber, FullReplication};
+use futures::StreamExt;
+use nomos_core::da::DaProtocol;
+use nomos_da::network::{adapters::libp2p::Libp2pAdapter, NetworkAdapter};
+use nomos_network::{backends::libp2p::Libp2p, NetworkService};
+use overwatch_derive::*;
+use overwatch_rs::{
+    overwatch::OverwatchRunner,
+    services::{
+        handle::{ServiceHandle, ServiceStateHandle},
+        relay::NoMessage,
+        state::*,
+        ServiceCore, ServiceData, ServiceId,
+    },
+    DynError,
+};
+use std::{path::PathBuf, time::Duration};
+
+#[derive(Args, Debug)]
+pub struct Disseminate {
+    // TODO: accept bytes
+    #[clap(short, long)]
+    data: String,
+    /// Path to the network config file
+    #[clap(short, long)]
+    network_config: PathBuf,
+    /// The data availability protocol to use. Defaults to full replication.
+    #[clap(flatten)]
+    da_protocol: DaProtocolChoice,
+    /// Timeout in seconds. Defaults to 120 seconds.
+    #[clap(short, long, default_value = "120")]
+    timeout: u64,
+}
+
+#[derive(Debug, Clone, Args)]
+pub struct FullReplicationSettings {
+    #[clap(long, default_value = "1")]
+    num_attestations: usize,
+}
+
+impl Disseminate {
+    pub fn run(&self) -> Result<(), Box<dyn std::error::Error>> {
+        tracing::subscriber::set_global_default(tracing_subscriber::FmtSubscriber::new())
+            .expect("setting tracing default failed");
```

Comment:
Shouldn't this be on the main function of the binary?

Reply:
Not sure everybody is interested in logs. For example, the complete interactive version of this command probably wants to use a different UI without printing node logs. Also, I'm using logs here to catch whatever the node is printing, a command that does not have a running node maybe does not even use tracing logs

Reply:
Sure, makes sense, thanks!